### PR TITLE
Remove omniauth-facebook

### PR DESCRIPTION
**Pull request by piyushswain**
This PR removes `omniauth-facebook` as one of the dependencies of this gem.

This has been done as `omniauth-facebook` doesn't work correctly with the `omniauth` versions >=2.0. This was forcing us to use older versions of `omniauth` for this gem.

[Issue#104](https://github.com/solidusio-contrib/solidus_social/issues/104)

### CODE REVIEW

File: app/assets/stylesheets/spree/frontend/fontello.css [hunk 1]
```diff
@@ -61,4 +61,4 @@
 .icon-spree-google-oauth2:before { content: '\e806'; } /* '' */
 .icon-spree-google-oauth2-circled:before { content: '\e807'; } /* '' */
 .icon-spree-amazon:before { content: '\e808'; } /* '' */
-.icon-spree-amazon-circled:before { content: '\e808'; } /* '' */
\ No newline at end of file
+.icon-spree-amazon-circled:before { content: '\e808'; } /* '' */
```
Comment (waiting-for-dev) [inline app/assets/stylesheets/spree/frontend/fontello.css]: We can remove all the references to amazon/facebook/twitter from this file, right?

File: spec/features/spree/sign_in_spec.rb [hunk 2]
```diff
@@ -1,55 +0,0 @@
-# frozen_string_literal: true
```
Comment (waiting-for-dev) [inline spec/features/spree/sign_in_spec.rb:1]: We shouldn't remove this spec. It used Facebook as an example but tested the generic sign-in process. We need to use another provider as an example now.

**Review by waiting-for-dev — COMMENTED** [review]
Thanks, @piyushswain. I left two comments.

File: app/assets/stylesheets/spree/frontend/fontello.css (hunk 1 above)
Comment (piyushswain) [reply app/assets/stylesheets/spree/frontend/fontello.css]: Removed here [4fb5b3d](https://github.com/solidusio-contrib/solidus_social/commit/4fb5b3dd6b0041caae55915836b3c5666fd83ed9)

File: spec/features/spree/sign_in_spec.rb (hunk 2 above)
Comment (piyushswain) [reply spec/features/spree/sign_in_spec.rb:1]: @waiting-for-dev I now think the problem is with the `omniauth` gem itself.

The sign_in_specs fail for `omniauth-facebook`, `omniauth-github` and `omniauth-google-oauth2` when `ominauth` is >=2.0, but the sign_in_spec passes for all when `omniauth` version in <2.0 .i.e 1.9.1

Comment (waiting-for-dev) [reply spec/features/spree/sign_in_spec.rb:1]: I see. Do you think it's easy to fix?

**Pull request closed without merging**